test/k8sT/DatapathConfiguration.go

```diff
@@ -142,33 +141,45 @@ var _ = Describe("K8sDatapathConfig", func() {
 				"global.bpf.monitorFlags":       "syn",
 				"global.debug.enabled":          "false",
 			})
-			monitorOutput, targetIP := monitorConnectivityAcrossNodes(kubectl, monitorLog)
-
-			By("Checking that exactly one ICMP notification in each direction was observed")
-			expEgress := fmt.Sprintf("ICMPv4.*DstIP=%s", targetIP)
-			expEgressRegex := regexp.MustCompile(expEgress)
-			egressMatches := expEgressRegex.FindAllIndex(monitorOutput, -1)
-			Expect(len(egressMatches)).To(Equal(1), "Monitor log contained unexpected number of egress notifications matching %q", expEgress)
-
-			expIngress := fmt.Sprintf("ICMPv4.*SrcIP=%s", targetIP)
-			expIngressRegex := regexp.MustCompile(expIngress)
-			ingressMatches := expIngressRegex.FindAllIndex(monitorOutput, -1)
-			Expect(len(ingressMatches)).To(Equal(1), "Monitor log contained unexpected number of ingress notifications matching %q", expIngress)
-
-			By("Checking the set of TCP notifications received matches expectations")
-			// | TCP Flags | Direction | Report? | Why?
-			// +===========+===========+=========+=====
-			// | SYN       |    ->     |    Y    | monitorFlags=SYN
-			// | SYN / ACK |    <-     |    Y    | monitorFlags=SYN
-			// | ACK       |    ->     |    N    | monitorFlags=(!ACK)
-			// | ACK       |    ...    |    N    | monitorFlags=(!ACK)
-			// | ACK       |    <-     |    N    | monitorFlags=(!ACK)
-			// | FIN       |    ->     |    Y    | monitorAggregation=medium
-			// | FIN / ACK |    <-     |    Y    | monitorAggregation=medium
-			// | ACK       |    ->     |    Y    | monitorAggregation=medium
-			egressPktCount := 3
-			ingressPktCount := 2
-			checkMonitorOutput(monitorOutput, egressPktCount, ingressPktCount)
+			monitorRes, monitorCancel, targetIP := monitorConnectivityAcrossNodes(kubectl)
+
+			var monitorOutput []byte // Accumulate output here over time
+			body := func() bool {
+				monitorOutput = append(monitorOutput, monitorRes.CombineOutput().Bytes()...)
```

Comment:
won't this end up appending the same beginning of output over and over again, as we are not consuming the bytes that were already read. This may be ok if we just want to find "bad" lines, but I think it would be enough to just `monitorOutput = monitorRes.CombineOutput().Bytes()`. The same applies to all similar lines below (196, 890 in Policies.go).

Reply:
You're right. For some reason I thought that there was a .Read call on a bytes.Buffer somewhere but that doesn't seem to be the case. I've switch it out. A bit less code too! I realised that we already have a WaitUntilMatch already on CmdRes.